### ISSUE TITLE
mockReset doesn't play nicely with jest-when

### ISSUE THREAD

**Issue by strottos** [timkindberg/jest-when#44]
Had a bit of bother with using `jest-when` this afternoon. Basically the following code triggers the issue:
```
const { when } = require('jest-when')
const testFn = jest.fn()

describe('testing', () => {
  afterEach(() => {
    testFn.mockReset()
  })

  test('test1', () => {
    when(testFn)
      .calledWith('test')
      .mockReturnValueOnce(1)
      .mockReturnValueOnce(2)
      .mockReturnValueOnce(3)
    expect(testFn('test')).toBe(1)
    expect(testFn.mock.calls.length).toBe(1)
  })

  test('test2', () => {
    when(testFn).calledWith('test').mockReturnValueOnce(1)
    expect(testFn('test')).toBe(1)
    expect(testFn.mock.calls.length).toBe(1)
  })
})
```

So in the code above I setup two tests and run them with `jest`. I set it so that the function I'll be mocking (a dummy function `testFn`) is reset by `.mockReset` in the `afterEach` so I know setup from one test won't bleed into the other.

I setup `test1` to have three values that get returned on calls but I only use one of them. After `test1` has ran I expect `test2` to not have anything associated with the mock. However if you run the above, rather than getting `1` as I expect I get `2` from `test1`. This doesn't happen if I don't add further return values, then I get nothing (as expected).

I can get around this by adding the following line, however I shouldn't have to and this caused me a lot of problems tracking down errors this afternoon.
```
  afterEach(() => {
    testFn.mockReset()
    testFn.__whenMock__ = undefined
  })
```

**Comment by strottos** [comment]
Another user side workaround would be running `verifyAllWhenMocksCalled` after the tests too. Though whilst the above implies that test1 is broken, what shouldn't happen is test2 breaks as a result of a broken test1.

**Comment by strottos** [comment]
Or as I've just learned I can call `resetAllWhenMocks` in the afterEach too which is what I'll do. It's possible you can't get this to play nicely with `jest`'s `mockReset` function in which case this will have to do (and is possibly why it's there). If this isn't possible feel free to close (but if it is it would be better).

**Comment by timkindberg** [comment]
This is true, the mockReset() doesn't quite work with how I've coded jest-when, I'm not sure how easy it would be to have it work. That is why we added `resetAllWhenMocks`, I hope that is sufficient.

**Comment by strottos** [comment]
Yeah makes sense having looked at it since. Thanks for looking anyway, no problem for closing 👍

**Comment by timkindberg** [comment]
Actually I think this is a legit bug... reopening.

**Comment by timkindberg** [comment]
Fixed in v2.7.2
